enhancements/kubelet/user-namespaces-support.md

```diff
@@ -0,0 +1,454 @@
+
+Specifically, both of these relate to how a cluster would be effected by the SCCs being created on the cluster, but the feature gate not being enabled (and thus `UserNamespaceLevel`
+is not being enforced).
+
+- `restricted-v3` SCC will be allowed for all users by default.
+    - Since it is identical to `restricted-v2` with the exception of `UserNamespaceLevel`, `UserNamespaceLevel` being filtered out because an apiserver doesn't recognize it anymore
+      (because the feature is no longer enabled) will allow it to be treated just like `restricted-v2` safely.
+- `nested-container` SCC will not be enabled for users by default.
+    - However, a namespace admin may enable it for a namespace, and then the cluster admin may trigger a downgrade, which would cause the issue spelled out above, but worse.
+    - Since the SCC allows users access to `runAsUser: RunAsAny`, the `nested-container` SCC may be enabled on a namespace that gives higher privileges than anticipated without `UserNamespaceLevel: RequirePodLevel`
+    - Thus, there should be documentation about 4.18 warning admins to check whether any pods are given access to `nested-container` SCC before downgrading.
+        - A script will be provided, to comb through the pods and notify if there are pods that have the labels signifying they're using the `nested-container` SCC.
+        - If there are such pods, these pods should be audited and potentially removed before downgrading.
+
+Downgrade of the kubelet will be handled by the section below.
+
+In the future, this feature will have the TechPreviewNoUpgrade flag removed, at which point all supported Kubelets and apiservers will
+be aware of the feature gate and attempt to create a pod with a user namespace.
+
+## Version Skew Strategy
+
+In Kubernetes and Openshift, a version skew of n-3 between the kubelet and apiserver is supported. The key consideration in version skew:
+if the kube-apiserver believes the cluster supports user namespaces, will every supported kubelet create a pod with a user namespace?
+
+Unfortunately, as described in the `MinimalKubeletVersion in the apiserver` section, this cannot be relied upon because we cannot ensure the kubelet
+will have the feature gate enabled. An operative part of this feature is relaxing validation done on kube-apiservers for pods with a user namespace,
+but if the apiserver cannot trust the kubelet to fail to create a pod if the feature isn't enabled, it cannot trust the kubelet with relaxed validation.
+
+The `MinimalKubeletVersion` feature is to fix this problem. If all apiservers support this field upon GA, then a cluster admin can set the field and ensure
+their kubelets are new enough to certainly support user namespaces. This frees the apiserver to relax validation and GA these SCC fields.
+
+This feature is not required for TechPreview, but is required for GA if we do not want to wait until 1.33/4.20 to GA this feature (the point where the oldest
```

Comment:
FeatureGate is still not enabled for kubelets in 1.31/4.18, so if MinimalKubeletVersion is not developed, it looks like 4.22 is the earliest possible.  If MinimalKubeletVersion is developed, the earliest is 4.18.

Reply:
why 4.22?

Reply:
> why 4.22?

Because as it currently stands 1.31/4.18 kubelets won't correctly honor the field.  4.19 becomes the first level where it might be properly honored and we have 4.19+3.